src/pages/ConnectWallet/ConnectWallet.tsx

```diff
@@ -11,22 +14,79 @@ import OrbsStatic from 'assets/orbs-static.png'
 import { FoxIcon } from 'components/Icons/FoxIcon'
 import { Page } from 'components/Layout/Page'
 import { RawText, Text } from 'components/Text'
-import { ActionTypes, WalletActions } from 'context/WalletProvider/WalletProvider'
+import { ActionTypes, InitialState, WalletActions } from 'context/WalletProvider/WalletProvider'
 import { useQuery } from 'hooks/useQuery/useQuery'
 import { colors } from 'theme/colors'
 
-type NoWalletProps = {
+import { KeyManager, SUPPORTED_WALLETS } from '../../context/WalletProvider/config'
+
+type WalletProps = {
+  state: InitialState
   dispatch: Dispatch<ActionTypes>
-  hasWallet: boolean
 }
 
-export const ConnectWallet = ({ dispatch, hasWallet }: NoWalletProps) => {
+export const ConnectWallet = ({ state, dispatch }: WalletProps) => {
+  const hasWallet = Boolean(state.walletInfo?.deviceId)
   const history = useHistory()
   const translate = useTranslate()
   const query = useQuery<{ returnUrl: string }>()
   useEffect(() => {
     hasWallet && history.push(query?.returnUrl ? query.returnUrl : '/dashboard')
-  }, [history, hasWallet, query.returnUrl])
+    if (
+      !(
+        localStorage.hasOwnProperty('walletId-cypress') &&
+        localStorage.hasOwnProperty('walletPwd-cypress')
+      )
+    ) {
+      return
+    }
+    const walletId = localStorage.getItem('walletId-cypress') || ''
+    const walletPwd = localStorage.getItem('walletPwd-cypress') || ''
+    ;(async () => {
+      // Here goes the programmatic login in case we are testing
```

Comment:
`// Programmatic login for Cypress tests`

Reply:
Fixed.